proof/refine/ARM/Invariants_H.thy

```diff
@@ -250,16 +250,24 @@ abbreviation scs_of' :: "kernel_state \<Rightarrow> obj_ref \<Rightarrow> sched_
 abbreviation scReplies_of :: "kernel_state \<Rightarrow> obj_ref \<Rightarrow> obj_ref option" where
   "scReplies_of s \<equiv> scs_of' s |> scReply"
 
-definition tcb_of' :: "kernel_object \<Rightarrow> tcb option" where
-  "tcb_of' kobj \<equiv> (case kobj of KOTCB tcb \<Rightarrow> Some tcb
-                               | _         \<Rightarrow> None)"
+abbreviation tcb_of' :: "kernel_object \<Rightarrow> tcb option" where
+  "tcb_of' \<equiv> projectKO_opt"
 
 abbreviation tcbs_of' :: "kernel_state \<Rightarrow> obj_ref \<Rightarrow> tcb option" where
   "tcbs_of' s \<equiv> ksPSpace s |> tcb_of'"
 
 abbreviation tcb_scs_of' :: "kernel_state \<Rightarrow> obj_ref \<Rightarrow> obj_ref option" where
   "tcb_scs_of' s \<equiv> tcbs_of' s |> tcbSchedContext"
```

Comment:
This is something existing, but I think we want to rename this to `tcbSCs_of`.

Reply:
Agreed, but can we leave that for another time? I would like to get this merged so I can continue with `sendIPC_corres`

Reply:
That is understandable, but, if you actually see that we currently have about five occurrences of this in the MCS Refine, where as this PR will add 30+ of them. So, I think it would make sense to change this here....